### PR TITLE
NHWC support in cuDNN BatchNorm & Conv2d

### PR DESCRIPTION
**Pull request by jjsjann123**
This reverts the 9a9bb448ee49a1493f22bbbeed4af92b1364fce9

Fixing the broken case which reverts the previous commit.
details about fix:
	modified:   aten/src/ATen/native/Convolution.cpp

called contiguous on 3D input tensor. This avoids the code path to accidentally
recognize the input as channel_last stride, due to unsqueezing of permuted 3d
tensor.

### CODE REVIEW

**Comment by jjsjann123** [comment]
The 3 line fix is here: https://github.com/pytorch/pytorch/pull/29361/files#diff-57ac615408468d3c7a461e505581bea3R568-R571

The previous failing case is actually still a legit nhwc case in cudnn. It's failing some trivial dimension checks. I'm working with cudnn team on fixing that.
Nevertheless it's undesired behavior due to my oversight earlier. The proper behavior would be to not accidentally use NHWC on 3d tensor (until we extend channel_last support to arbitrary dimension tensors)

**Review by VitalyFedyunin — APPROVED** [review]
